modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/cloud/JsonPathDiscovery.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2012-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.cloud
+
+import cats.Monad
+import cats.implicits._
+import com.snowplowanalytics.snowplow.rdbloader.DiscoveryAction
+import com.snowplowanalytics.snowplow.rdbloader.common.Common.toSnakeCase
+import com.snowplowanalytics.snowplow.rdbloader.common.cloud.BlobStorage
+import com.snowplowanalytics.snowplow.rdbloader.discovery.DiscoveryFailure
+import com.snowplowanalytics.snowplow.rdbloader.discovery.DiscoveryFailure.JsonpathDiscoveryFailure
+import com.snowplowanalytics.snowplow.rdbloader.discovery.ShreddedType.Info
+import com.snowplowanalytics.snowplow.rdbloader.dsl.Cache
+
+trait JsonPathDiscovery[F[_]] {
+  def discoverJsonPath(jsonpathAssets: Option[BlobStorage.Folder], shreddedType: Info): DiscoveryAction[F, BlobStorage.Key]
+}
+
+object JsonPathDiscovery {
```

Comment:
I've placed `JsonPathDiscovery` and `LoadAuthService` under `loader` module since they are only used in the loader. It doesn't look good to have `cloud` packages in both `common` and `loader` modules but this seems best way to me. Let me know if you have other opinions.